flink-python/src/main/java/org/apache/flink/streaming/api/runners/python/beam/BeamPythonFunctionRunner.java

```diff
@@ -681,7 +686,14 @@ public void onCompleted(BeamFnApi.ProcessBundleResponse response) {
 
     private TimerReceiverFactory createTimerReceiverFactory() {
         BiConsumer<Timer<?>, TimerInternals.TimerData> timerDataConsumer =
-                (timer, timerData) -> timerRegistration.setTimer((byte[]) timer.getUserKey());
+                (timer, timerData) ->
+                        environment
+                                .getMainMailboxExecutor()
```

Comment:
`environment.getMainMailboxExecutor()` has a side effect that it will create MainMailboxExecutor instance each time called. What about making MainMailboxExecutor an instance variable?

Reply:
I believe the mainMailboxExecutor is set once for the StreamTask here https://github.com/apache/flink/blob/60497e7dfddf27b25b80b913ef259010e97569aa/flink-runtime/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java#L458. And when Environment.getMainMainboxExecutor() is called here https://github.com/apache/flink/blob/60497e7dfddf27b25b80b913ef259010e97569aa/flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/RuntimeEnvironment.java#L376, it only returns the instance variable directly. I did not see any side effect in the code, or do I miss anything? Thanks